### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by MenendezON**
This pull request concerns the different stages of our Budget app project:
- [ ] Set up the repository and tools.
- [ ]  Create the models according to the ERD diagram. Don't forget about validations and tests.
- [ ]  Implement the authentication and authorization.
- [ ]  Create the page to add a category.
- [ ]  Create the home page.
- [ ]  Create the page to add a transaction.
- [ ]  Create the transactions page.
- [ ]  Make sure your app is tested adequately.
- [ ]  Deploy the project, and test for final details.
- [ ]  Record a video for your project.
- [ ]  Create a good README and PR description.

### CODE REVIEW

File: app/views/categories/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  <img src="https://cdn-icons-png.flaticon.com/512/8541/8541407.png" alt="No trades found">
+<% else %>
+    <div class="cards">
+      <% @trades.each do |trade| %>
+      <div class="card">
+        <%= image_tag(@category.icon, alt: "Icon pic", class: "left") %>
+        <div class="right">
+          <div class="sub-right">
+            <p class="trade-name"><%= trade.name %></p>
+            <p class="trade-amount">$<%= trade.amount %></p>
```
Comment (Shedrack-Sunday) [inline app/views/categories/show.html.erb:19]: - [ ] Whats expected is the amount of transaction in a category, not the price. Lets assume you have 4 transation, 6 should be there not the prices of the 6.

### Instructions:
image

### Your view:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @MenendezON        👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with Rails features ✔️
- Great work on the page implementation👏 
 

# - Required Changes  🌵  

- [ ] Kindly note the changes i suggested and pointed out in your code. There are a lot, from the README to the pages, and i advice you to go over it carefully  
- [ ] For your tests, kindly make sure all are  passing and each method is tested 
# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/features/categories_spec.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    it 'renders category name' do
+      expect(page).to have_content(@category.name)
+    end
+
+    it 'renders category icon' do
+      find("img[src='#{@category.icon}']")
+    end
+
+    it 'renders category\'s total expenses' do
+      expect(page).to have_content(@category.expenses.sum(:amount))
```
Comment (OyePriscilla) [inline spec/features/categories_spec.rb:22]: - [ ] As stated earlier, some of the tests that you've written are failing at the moment. **The only one that passed succesfully were tests written in `/models` directory**, kindly work around this in order to fulfil the project requirement. 🙏

File: README.md [hunk 3]
```diff
@@ -1 +1,137 @@
...
+- **The desing is for mobile first approach**
+- **For handling user's log and register the App uses device gem**
+
+### Live demo <a name="live-demo">
+[Budget App](https://capstone-budgetapp.onrender.com)
+</a>
+
+### Presentation Video <a name="presentation">
+[Budget App](https://www.loom.com/share/231cd971f0094169b35112f8a25abbb2?sid=ba4041f9-58dd-42f3-a85f-18730283a6c8)
+</a>
```
Comment (OyePriscilla) [inline README.md:52]: - [ ] Kindly please indicate the piece of code or something that you find interesting in the process of building the app in your video presentation as well. 👍 

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @MenendezON 👏 👏 👏 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Removed commented code ✔️ 
Added live demo and video presentation page ✔️ 
Implemented categories and transaction pages as expected ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi @MenendezON 🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [ ] Although you performed exceptionally well, there are currently some tests that are not passing. Please review those tests and implement the required modifications.

Screenshot from 2023-12-01 11-15-59

- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @MenendezON, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ All tests are passing

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**